src/connectionconfig/connectionDialogWebviewController.ts

```diff
@@ -1,655 +1,660 @@
-			this.clearFormError();
-			this.state.connectionProfile = payload.connection;
-			await this.updateItemVisibility();
-			await this.handleAzureMFAEdits('azureAuthType');
-			await this.handleAzureMFAEdits('accountId');
-			return state;
-		});
-
-		this.registerReducer('connect', async (state) => {
-			this.clearFormError();
-			this.state.connectionStatus = ApiStatus.Loading;
-			this.state.formError = '';
-			this.state = this.state;
-
-			const usedFields = new Set<keyof IConnectionDialogProfile>(this.getActiveFormComponents().filter(c => !c.hidden).map(c => c.propertyName));
-
-			Object.keys(this.state.connectionFormComponents.advancedComponents).forEach(group => {
-				this.state.connectionFormComponents.advancedComponents[group].forEach(c => {
-					if (!c.hidden) {
-						usedFields.add(c.propertyName);
-					}
-				});
-			});
-
-			// Clear unused fields (anything that isn't visible due to form selections and isn't an advanced option)
-			Object.keys(this.state.connectionProfile).forEach(optionName => {
-				if (!usedFields.has(optionName as keyof IConnectionDialogProfile)) {
-					// eslint-disable-next-line @typescript-eslint/no-explicit-any
-					(this.state.connectionProfile[optionName as keyof IConnectionDialogProfile] as any) = undefined;
-				}
-			});
-
-			// Perform final validation of all inputs
-			const errorCount = await this.validateFormComponents();
-			if (errorCount > 0) {
-				this.state.connectionStatus = ApiStatus.Error;
-				return state;
-			}
-
-			try {
-				const result = await this._mainController.connectionManager.connectionUI.validateAndSaveProfileFromDialog(this.state.connectionProfile as any);
-				if (result?.errorMessage) {
-					this.state.formError = result.errorMessage;
-					this.state.connectionStatus = ApiStatus.Error;
-					return state;
-				}
-				if (this._connectionToEditCopy) {
-					await this._mainController.connectionManager.getUriForConnection(this._connectionToEditCopy);
-					await this._objectExplorerProvider.removeConnectionNodes([this._connectionToEditCopy]);
-					await this._mainController.connectionManager.connectionStore.removeProfile(this._connectionToEditCopy as any);
-					await this._objectExplorerProvider.refresh(undefined);
-				}
-				await this._mainController.connectionManager.connectionUI.saveProfile(this.state.connectionProfile as any);
-				const node = await this._mainController.createObjectExplorerSessionFromDialog(this.state.connectionProfile);
-				await this._objectExplorerProvider.refresh(undefined);
-				await this.loadRecentConnections();
-				this.state.connectionStatus = ApiStatus.Loaded;
-				await this._mainController.objectExplorerTree.reveal(node, { focus: true, select: true, expand: true });
-				await this.panel.dispose();
-			} catch (error) {
-				this.state.connectionStatus = ApiStatus.Error;
-				return state;
-			}
-			return state;
-		});
-	}
-}
+/*---------------------------------------------------------------------------------------------

```

Comment:
not sure what's going on with the diff for this file...  The highlight is that I reorganized how form components are stored in the state (one record for everything), and how the different modes/views reference the specific form components (by propertyName instead of duplicating the entire component.

The idea is that this makes the connection dialog more composable without keeping state copied around.